packages/core/src/components/modal/modal.tsx

```diff
@@ -58,24 +59,24 @@ export class AtomModal {
     }
   }
 
-  private handleDidDimiss = () => {
+  private readonly handleDidDimiss = () => {
     this.atomDidDismiss.emit(this.modal)
   }
 
-  private handleDidPresent = () => {
+  private readonly handleDidPresent = () => {
     this.atomDidPresent.emit(this.modal)
   }
 
-  private handleCloseClick = async () => {
+  private readonly handleCloseClick = async () => {
     this.atomCloseClick.emit(this.modal)
     this.modal.close()
   }
 
-  private handleSecondaryClick = () => {
+  private readonly handleSecondaryClick = () => {
     this.atomSecondaryClick.emit(this.modal)
   }
 
-  private handlePrimaryClick = () => {
+  private readonly handlePrimaryClick = () => {
     this.atomPrimaryClick.emit(this.modal)
   }
```

Comment:
sonar wasnt well configured, so now that is checking code on the right way, it throws an error of the functions and properties never being reassigned, so they can be marked as readonly